core/modules/replica/WorkerSqlRequest.cc

```diff
@@ -274,21 +296,52 @@ string WorkerSqlRequest::_query(database::mysql::Connection::Ptr const& conn) co
         default:
             throw invalid_argument(
                     "WorkerSqlRequest::" + string(__func__) +
-                    "  unsupported request type: " + ProtocolRequestSql_Type_Name(requestType));
+                    "  unsupported request type: " + ProtocolRequestSql_Type_Name(_request.type()));
     }
 }
 
 
-void WorkerSqlRequest::_setResponse(database::mysql::Connection::Ptr const& conn) {
+string WorkerSqlRequest::_batchQuery(database::mysql::Connection::Ptr const& conn,
+                                     string const& table) const {
+
+    string const databaseTable = conn->sqlId(_request.database()) + "." + conn->sqlId(table);
+
+    switch (_request.type()) {
+        case ProtocolRequestSql::DROP_TABLE:
+            return "DROP TABLE IF EXISTS " + databaseTable;
+
+        case ProtocolRequestSql::DROP_TABLE_PARTITION:
+            return "ALTER TABLE " + databaseTable + " DROP PARTITION IF EXISTS "
+                   + conn->sqlId("p" + to_string(_request.transaction_id()));
```

Comment:
Looks like hardcoded convention for partition naming, should be defined in one place.